lib/wlib/comm/PacketConstructor.h

```diff
@@ -0,0 +1,142 @@
+/** @file PacketConstructor.h
+ *  @brief Creates a packet for sending data from the slave systems to the main computer.
+ *
+ *  PacketConstructor is an abstract class. Therefore, it cannot be instantiated.
+ *  Subclasses include SensorPacket, CommandPacket, StatePacket, and LogPacket.
+ *
+ *  @author Heather D'Souza (heathkd)
+ *  @date November 7, 2017
+ *  @bug No known bugs.
+ */
+
+#ifndef EMBEDDEDCPLUSPLUS_PACKETCONSTRUCTOR_H
+#define EMBEDDEDCPLUSPLUS_PACKETCONSTRUCTOR_H
+
+#include <string>
+#include "../stl/Bitset.h"
+
+using namespace wlp;
+
+/**
+ *  @brief Blueprint for a packet of type Bitset. Cannot be instatiated.
+*/
+class PacketConstructor {
+
+public:
+    PacketConstructor();
+    PacketConstructor(float *p_data); // Create a new Packet
+    ~PacketConstructor(); // Destroy this packet freeing all dynamically allocated memory.
+
+    //Helper Methods
+    /** @brief Sets the start and end bits.
+     *
+     * Each packet should begin and end with a '1' bit so that
+     * parser can distinguish the end of one packet and the
+     * beginning of another.
+     *
+     * @return Void.
+    */
+    void setStartAndEnd();
+    /** @brief Sets the type and name of the packet.
+     *
+     * There are 4 types of packets: sensor, command, state, and log.
+     * This method sets the bit representation of these types and the name
+     * pertaining to the specific packet. This function must be defined in each
+     * subclass of PacketConstructor so that the child class may be instantiated.
+     *
+     * @return Void.
+    */
+    virtual void setTypeAndName()=0;
```

Comment:
Modern style is `= delete`

Reply:
Doesn't this allows you to hide the method but ``=0`` allows you to override it?

Reply:
You're right I lied